src/shared/helpers/resolver.js

```diff
@@ -0,0 +1,61 @@
+import _ from 'lodash';
+
+function resolveAllOfItem(node) {
+  const output = _.cloneDeep(node);
+  delete output.allOf;
+
+  const allOfItems = node.allOf;
+  for (let i = 0; i < allOfItems.length; i++) {
+    const item = allOfItems[i];
+
+    Object.keys(item).forEach(key => {
+      if (!output.hasOwnProperty(key)) {
+        output[key] = _.cloneDeep(item[key]);
+      } else if (key === 'properties') {
+        const properties = item[key];
+
+        Object.keys(properties).forEach(name => {
+          output.properties[name] = _.cloneDeep(properties[name]);
+        });
+      } else if (key === 'required') {
+        // Concatenate to existing list and remove duplicates
+        const requiredArray = _.uniq(output.required.concat(item[key]));
+        output.required = requiredArray.sort();
+      }
+    });
+  }
+
+  return output;
+}
+
+export function resolveAllOfRecursive(obj) {
```

Comment:
updated docblock.

It's intentionally not pure. It's more tricky to make it pure with a recursive function. The object is cloned before it gets to this function. IMHO, a function can either modify input or return some output, and not do both. This function does the first thing only.